lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1768,15 +1800,14 @@ class BuildPipelineRegs : public FuncOpPartialLoweringPattern {
         Value stageResult = stage.getResult(i);
         bool isIterArg = false;
         for (auto &use : stageResult.getUses()) {
+          unsigned operandNo = use.getOperandNumber();
           if (auto term =
-                  dyn_cast<staticlogic::PipelineTerminatorOp>(use.getOwner())) {
-            if (use.getOperandNumber() < term.iter_args().size()) {
-              WhileOpInterface whileOp(stage->getParentOp());
-              auto reg = getComponentState().getWhileIterReg(
-                  whileOp, use.getOperandNumber());
-              getComponentState().addPipelineReg(stage, reg, i);
-              isIterArg = true;
-            }
+                  dyn_cast<staticlogic::PipelineTerminatorOp>(use.getOwner());
+              term && operandNo < term.iter_args().size()) {
```

Comment:
IMO this is doing too much in the condition of an `if`. I don't know if this is LLVM coding standard, but if there is an assignment inside an if condition, I think it is best for this to be the only statement within an if condition (this is why there were nested ifs originally.

Reply:
Is there any meaningful change here or is this just style? Please commit style changes separately from meaningful changes.

Reply:
> but if there is an assignment inside an if condition, I think it is best for this to be the only statement within an if condition (this is why there were nested ifs originally.

I think C++17's if statement with initializer makes this read quite nicely. We really only care if `A && B` are true, and thus this feels more succinct. It seems odd to write this as nested `if`s. However, both are perfectly valid choices, and probably boil down to what we're accustomed to.

In any case, the style changes were probably me adding some debugging code. I'll fix these.

Reply:
I thought there was a coding standard in LLVM or the Google style guide about not doing anything more than a simple assignment, but I'm not finding it. This might be carrying over a Ruby convention I used... Anyway, I guess this is my somewhat curmudgeonly stance on writing dead simple C++ coming out again.